blaze/server/tests/test_server.py

```diff
@@ -621,3 +658,77 @@ def test_bad_add_payload(empty_server, serial):
         data=blob,
     )
     assert response1.status_code == 422
+
+
+@pytest.mark.parametrize('serial', all_formats)
+def test_odo_kwargs(test, serial):
+    expr = t.dumb
+    bad_query = {'expr': to_tree(expr)}
+
+    result = test.post(
+        '/compute',
+        headers=mimetype(serial),
+        data=serial.dumps(bad_query),
+    )
+    assert result.status_code == 500
+    assert b'return_df must be passed' in result.data
+
+    good_query = {
+        'expr': to_tree(expr),
+        'odo_kwargs': {
+            'return_df': odo(DumbResource.df, list),
+        },
+    }
+    result = test.post(
+        '/compute',
+        headers=mimetype(serial),
+        data=serial.dumps(good_query)
+    )
+    assert result.status_code == 200
+    data = serial.loads(result.data)
+    dshape = discover(DumbResource.df)
+    assert_dshape_equal(
+        datashape.dshape(data['datashape']),
+        dshape,
+    )
+    assert_frame_equal(
+        odo(data['data'], DataFrame, dshape=dshape),
+        DumbResource.df,
+    )
+
+
+@pytest.mark.parametrize('serial', all_formats)
+def test_compute_kwargs(test, serial):
+    expr = t.dumb.sort()
+    bad_query = {'expr': to_tree(expr)}
+
+    result = test.post(
+        '/compute',
+        headers=mimetype(serial),
+        data=serial.dumps(bad_query),
+    )
+    assert result.status_code == 500
+    assert b'return_df must be passed' in result.data
+
+    good_query = {
+        'expr': to_tree(expr),
+        'compute_kwargs': {
+            'return_df': odo(DumbResource.df, list),
```

Comment:
There's a subtlety here -- this is the same kv pair as for the `odo_kwargs` above which I didn't see at first; since it ends up hitting the `compute_down()` dispatch on line 62, it ultimately calls `dumb_to_df` and everything works out.